pandas/core/resample.py

```diff
@@ -1085,7 +1086,23 @@ def interpolate(
         """
         assert downcast is lib.no_default  # just checking coverage
         result = self._upsample("asfreq")
-        return result.interpolate(
+
+        # If the original data has timestamps which are not aligned with the
+        # target timestamps, we need to add those points back to the data frame
+        # that is supposed to be interpolated. This does not work with
+        # PeriodIndex, so we skip this case.
+        obj = self._selected_obj
+        is_period_index = isinstance(obj.index, PeriodIndex)
+
+        if not is_period_index:
+            final_index = result.index
+            missing_data_points_index = obj.index.difference(final_index)
+            if len(missing_data_points_index) > 0:
```

Comment:
in the opposite case where the difference is empty, i think a the sort_index and `.loc` below (which makes a copy) can be avoided

Reply:
@jbrockmendel Can you please clarify?  "in the opposite case where the difference is empty" --> then `missing_data_points_index` will be empty, so its length will be zero. In that case, the if-expression evaluates to `False`, so that the loc/sort is indeed not executed. Maybe I am overlooking something.